Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +902,76 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getAsString(sectionName: string, settingName: string, resource: any, defaultValue: string): string {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<string>(settingName);
```

Comment:
As a followup to @colengms's comment, can you check if this returns null or undefined when a bogus sectionName is passed in? The API says it shouldn't, but let's make sure.

Reply:
Sorry if I was unclear. I wasn't asking you to write code to check it. I was asking you to debug it and see. If the API doesn't return these things, there's no point in checking for them. I wanted to know if we needed to do anything special for that case here.

Reply:
It looks like we don't need to check for null or undefined. The API returns its own special object.

Reply:
Does it still return an object when the setting doesn't exist?

Reply:
I sent you a teams message with more details.

Reply:
> Yes but the object doesn't contain any properties.

I don't expect it to. I only expect it to have the interface methods. As long as calling those methods doesn't crash, I expect the validation to continue working as expected.